### ISSUE TITLE
Fetch Zip Archives

### ISSUE THREAD

**Issue by rgrinberg** [ocaml/dune#10104]
https://github.com/ocaml/dune/pull/10085 was a hotfix that did not implement the full extraction functionality supported by opam. One such missing feature is the support for zip archives. We should restore this feature.

We should also make sure we support tbz archives while we're at it.

**Comment by Leonidas-from-XIV** [comment]
The way I use `tar` is that it automatically decompresses the formats it knows about (the manpage says that `-a`/`--auto-compress` can be used to enable it however it doesn't state that it is always enabled which it seems to be on my system), so it should support gzip, bzip2, xz, lzip, lzma, lzop (at least on my system) without any issue.

Zip support does indeed need a separate tool.

**Comment by rgrinberg** [comment]
> so it should support gzip, bzip2, xz, lzip, lzma, lzop (at least on my system) without any issue.

I see. I wonder why opam tries to do the detection manually then.

**Comment by Leonidas-from-XIV** [comment]
That could be up to different `tar` implementations. Looks like e.g. `tar` in OpenBSD only supports gzip & compress and doesn't do autodetection [according to its manpage](https://man.openbsd.org/tar).

**Comment by Alizter** [comment]
This appears to have been fixed a while ago. @Leonidas-from-XIV anything else left in this issue or can we close?

**Comment by Leonidas-from-XIV** [comment]
No, I don't think so. Support for decompressing XZ consistently between platforms is tracked in #13715 and LZMA in #13716.
